pallets/parachain-staking/src/benchmarks.rs

```diff
@@ -2192,6 +2192,71 @@ benchmarks! {
 	verify {
 		assert_eq!(T::Currency::free_balance(&collator), original_free_balance + 50u32.into());
 	}
+
+	notify_inactive_collator {
+		use crate::{AtStake, CollatorSnapshot, AwardedPts};
+
+		// Blocks per-round must be greater than TotalSelected
+		Pallet::<T>::set_blocks_per_round(RawOrigin::Root.into(), 101u32)?;
+		Pallet::<T>::set_total_selected(RawOrigin::Root.into(), 100u32)?;
+
+		let mut candidate_count = 1u32;
+		let mut seed = USER_SEED;
+
+		// Create 100 collators
+		for i in 0..100 {
```

Comment:
The `notify_inactive_collator` extrinsic is highly dependent on the length of `<SelectedCandidates<T>>`. Since this is a big vec, it will have a big impact on proof size (and definitely a measurable impact on time spent decoding).

In your benchmark, you use a static `100` as the size of this vec. That might be safe for networks where there are fewer than 100 selected candidates, but it's unsafe for anything with more. It's also needlessly expensive to call for networks with fewer than 100.

One simple fix for that is to use `T::MaxCandidates::get()` instead of 100. This would make it use the runtime-configured value, which is the worst case. The problem with this is that the caller may overpay, although I think in practice most networks will indeed have this worst-case (they will have the max selected candidates).

The proper way to fix it would be to make the benchmark dependent on the size of the vec, e.g. replace `100` with a parameterized value and let the benchmarking system determine how much Weight to scale with the size of the vec. Then you can do one of two things:

* Add a "hint" parameter to the call so the user can specify how many candidates currently exist (bad UX because it's obnoxious)
* Or charge the max up front and then refund if applicable.

I would recommend at least replacing `100` with `T::MaxCandidates::get()` for now, as this is at least safe for any runtime config.

Reply:
The motivation for someone calling this is probably the small economic advantage they get for having one less collator eligible to make blocks, so it needs to be cheap enough to call this for that incentive to kick in.

Reply:
Actually, we can use `decode_len()` and optimize all of this away: https://paritytech.github.io/polkadot-sdk/master/frame_support/storage/types/struct.StorageValue.html#method.decode_len

Reply:
Yes I think we can change the number of collators in the benchmark to `T::MaxCandidates::get()` and also optimize the length read with `decode_len()`, to ensure we cover the correct scenario and we don't charge excessive weight for it.